public/app/features/alerting/unified/rule-list/DataSourceGroupLoader.tsx

```diff
@@ -0,0 +1,200 @@
+import { useMemo } from 'react';
+
+import { Alert } from '@grafana/ui';
+import { t } from 'app/core/internationalization';
+import { DataSourceRuleGroupIdentifier } from 'app/types/unified-alerting';
+import {
+  PromRuleDTO,
+  PromRuleGroupDTO,
+  RulerCloudRuleDTO,
+  RulerRuleGroupDTO,
+  RulesSourceApplication,
+} from 'app/types/unified-alerting-dto';
+
+import { alertRuleApi } from '../api/alertRuleApi';
+import { featureDiscoveryApi } from '../api/featureDiscoveryApi';
+import { prometheusApi } from '../api/prometheusApi';
+import { RULE_LIST_POLL_INTERVAL_MS } from '../utils/constants';
+import { hashRule } from '../utils/rule-id';
+import { getRuleName, isCloudRulerGroup } from '../utils/rules';
+
+import { DataSourceRuleListItem } from './DataSourceRuleListItem';
+import { RuleOperationListItem } from './components/AlertRuleListItem';
+import { AlertRuleListItemLoader } from './components/AlertRuleListItemLoader';
+import { RuleActionsButtons } from './components/RuleActionsButtons.V2';
+import { RuleOperation } from './components/RuleListIcon';
+import { matchRulesGroup } from './ruleMatching';
+
+const { useDiscoverDsFeaturesQuery } = featureDiscoveryApi;
+const { useGetGroupsQuery } = prometheusApi;
+const { useGetRuleGroupForNamespaceQuery } = alertRuleApi;
+
+export interface DataSourceGroupLoaderProps {
+  groupIdentifier: DataSourceRuleGroupIdentifier;
+  expectedRulesCount?: number;
+}
+
+export function DataSourceGroupLoader({ groupIdentifier, expectedRulesCount = 3 }: DataSourceGroupLoaderProps) {
```

Comment:
`DataSourceGroupLoader` tries to load both Prometheus and Ruler endpoints for the specified group. 
There is a pooling for Prometheus query, to update if there are any changes in progress